sharding/nodesSetup.go

```diff
@@ -224,15 +247,15 @@ func (ns *NodesSetup) InitialNodesPubKeysForShard(shardId uint32) ([]string, err
 }
 
 // InitialNodesInfoForShard - gets initial nodes info for shard
-func (ns *NodesSetup) InitialNodesInfoForShard(shardId uint32) ([]*NodeInfo, error) {
-	if ns.allNodesInfo[shardId] == nil {
-		return nil, ErrShardIdOutOfRange
+func (ns *NodesSetup) InitialNodesInfoForShard(shardId uint32) ([]*NodeInfo, []*NodeInfo, error) {
+	if ns.eligible[shardId] == nil {
+		return nil, nil, ErrShardIdOutOfRange
 	}
-	if len(ns.allNodesInfo[shardId]) == 0 {
-		return nil, ErrNoPubKeys
+	if len(ns.eligible[shardId]) == 0 {
+		return nil, nil, ErrNoPubKeys
 	}
 
-	return ns.allNodesInfo[shardId], nil
+	return ns.eligible[shardId], ns.waiting[shardId], nil
```

Comment:
The `if`s above only verify `ns.eligible`, ignoring `ns.waiting`. Is that intentional?

Reply:
no nodes in waiting list is not exactly an error for the initial setup, so intentional.